File: 3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.cpp
class Solution {
public:
    vector<int> resultArray(vector<int>& nums) {
        vector<int> arr1;
        vector<int> arr2;
        arr1.push_back(nums[0]);
        
        arr2.push_back(nums[1]);
        
        for(int i=2;i<nums.size();i++)
        {
            int n1=arr1.size();
            int n2=arr2.size();
            if(arr2[n2-1]<arr1[n1-1])
            {
                arr1.push_back(nums[i]);
            }
            else
            {
                arr2.push_back(nums[i]);
            }
        }
        for(int i:arr2)
        {
            arr1.push_back(i);
        }
        return arr1;
    }
};

File: Two-Pointers/3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.py
class Solution:
    def resultArray(self, nums: List[int]) -> List[int]:
        l1=list()
        l2=list()
        l1.append(nums[0])
        l2.append(nums[1])

        a=nums[0]
        b=nums[1]

        for i in range(2,len(nums)):
            if a>b:
                l1.append(nums[i])
                a=nums[i]
                
            else:
                l2.append(nums[i])
                b=nums[i]
                

        return l1+l2
        